CMakeLists.txt

```diff
@@ -138,6 +138,27 @@ option(GOOGLE_CLOUD_CPP_ENABLE_FIRESTORE
 option(GOOGLE_CLOUD_CPP_ENABLE_PUBSUB "Enable building the Pub/Sub library." ON)
 option(GOOGLE_CLOUD_CPP_ENABLE_GENERATOR "Enable building the generator." ON)
 
+set(GOOGLE_CLOUD_CPP_LIBRARIES
+    "bigquery;bigtable;spanner;storage;firestore;pubsub;generator")
+
+# The default list of libraries to build. These can be overridden by the user by
+# passing a comma-separated list, i.e
+# `-DGOOGLE_CLOUD_CPP_ENABLE=spanner,storage`.
+set(GOOGLE_CLOUD_CPP_ENABLE
+    ${GOOGLE_CLOUD_CPP_LIBRARIES}
+    CACHE STRING "The list of libraries to build.")
+
+string(REPLACE "," ";" GOOGLE_CLOUD_CPP_ENABLE "${GOOGLE_CLOUD_CPP_ENABLE}")
+
+# Remove any library that's been disabled from the list.
+foreach (library ${GOOGLE_CLOUD_CPP_LIBRARIES})
+    string(TOUPPER "${library}" _library)
+    set(_library "GOOGLE_CLOUD_CPP_ENABLE_${_library}")
+    if (NOT ${_library})
+        list(REMOVE_ITEM GOOGLE_CLOUD_CPP_ENABLE ${library})
```

Comment:
Note: Here's the right place for us to add a warning message that these per-library enable flags should be avoided in favor of `GOOGLE_CLOUD_CPP_ENABLE`.  I think we may want to add a message for that.

@coryan WDYT?

Reply:
Fixed.